source/simulator/assembly.cc

```diff
@@ -608,17 +608,15 @@ namespace aspect
   template <int dim>
   void Simulator<dim>::assemble_stokes_system ()
   {
-    if (!assemble_newton_stokes_system)
-      computing_timer.enter_section ("   Assemble Stokes system");
-    else if (assemble_newton_stokes_matrix)
-      {
-        if (newton_handler->get_newton_derivative_scaling_factor() == 0)
-          computing_timer.enter_section ("   Assemble Stokes system picard");
-        else
-          computing_timer.enter_section ("   Assemble Stokes system newton");
-      }
-    else
-      computing_timer.enter_section ("   Assemble Stokes system rhs");
+    TimerOutput::Scope timer (computing_timer,
+                              (!assemble_newton_stokes_system ?
+                               "   Assemble Stokes system" :
+                               (assemble_newton_stokes_matrix ?
+                                (newton_handler->get_newton_derivative_scaling_factor() == 0 ?
```

Comment:
this is quite an ugly monstrosity. Would it be simpler if you leave the if block and assign a const char* instead?

Reply:
Yes, that place is ugly. I would need to create another variable with the same scope as the timer section for this. I can do that if you want me to.

Reply:
Shame that we can not use a lambda yet. I don't feel strongly about it. Either way, go ahead and merge yourself.

Reply:
Yes, a lambda would be nice here...